pkg/reconciler/apis/apibinding/apibinding_indexes.go

```diff
@@ -86,3 +87,15 @@ func indexAPIBindingsByIdentityGroupResourceFunc(obj interface{}) ([]string, err
 func IdentityGroupResourceKeyFunc(identity, group, resource string) string {
 	return fmt.Sprintf("%s/%s/%s", identity, group, resource)
 }
+
+const indexCustomResourceDefinitionByWorkspace = "customResourceDefinitionByWorkspace"
```

Comment:
it must be a unique name, by controller. We do `controllerName + "ByWorkspace"` elsewhere, until a more generic variant lands. Compare https://github.com/kcp-dev/apimachinery/pull/33 and its discussion.